src/components/App.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+import {Component, CSSProperties} from 'react';
+import {FloatingInput, Icon, IconButton} from 'jsc-react-ui';
+import {CardList} from './../components/CardList';
+import {CardForm} from './../components/CardForm';
+import {Note} from './../models/Note';
+
+const styles = {
+  container: {
+    padding: 10,
+  },
+  actionContainer: {
+    display: 'flex',
+    justifyContent : 'space-between',
+    margin: '25px 25px 0 25px',
+  } as CSSProperties,
+  filter: {
+    width: 246,
+  },
+  icon: {
+    margin: '0 14px 0 18px',
+  },
+  iconButton : {
+    backgroundColor: '#F1C40F',
+  },
+};
+
+interface State {
+  notes: Note[];
+  filter: string;
+  modalOpen: boolean;
+}
+
+export class App extends Component<{}, State> {
+
+  state: State = {
+    notes: [],
+    filter: '',
+    modalOpen: false,
+  };
+
+  componentDidMount() {
+    fetch('api/notes.json')
+      .then((response) => response.json())
+      .then((responseJson) => {
+        this.setState({notes: responseJson});
+      });
```

Comment:
You can deal with promises in a beautiful way using [async/await](https://basarat.gitbooks.io/typescript/docs/async-await.html). It's like programming synchronously.

```typescript
async componentDidMount() {
  const notes = await fetch('api/notes.json')
    .then(response => response.json());
  // ...
}
```

Reply:
Legal